content/artifacts/highlight-active-artifact-planets/plugin.js

```diff
@@ -0,0 +1,173 @@
+// Display the planet with active artifact, specially for cannon.
+
+import { 
+  BiomeNames
+} from "https://cdn.skypack.dev/@darkforest_eth/types";
+
+import {
+  coords
+} from 'https://plugins.zkga.me/utils/utils.js';
+
+import {
+  ArtifactType,
+  ArtifactTypeNames
+} from "https://cdn.skypack.dev/@darkforest_eth/types";
+
+const ARTIFACT_TYPES = Object.values(ArtifactType).filter((type) => type !== ArtifactType.Unknown).map((type) => ({ value: type, text: ArtifactTypeNames[type] }));
+
+
+function getPlayerColor(ethAddress) {
+  return df.getProcgenUtils().getPlayerColor(ethAddress);
+}
+
+
+function drawArc(ctx, planet, color, artifactTypes){
+  if(!planet) {
+    return false;
```

Comment:
does this happen sometimes?

Reply:
I have removed this redundant code , thanks.